pkg/version/version.go

```diff
@@ -37,6 +38,7 @@ func LogVersionToLogger(log logd.Logger) {
 		"buildDate", BuildDate,
 		"goVersion", runtime.Version(),
 		"platform", fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
+		"fips140", fmt.Sprintf("FIPS 140-3 Mode Enabled: %v", fips140.Enabled()),
```

Comment:
Don't know where to put this, but shouldn't we put our fips image into "fips only mode"?

If I understand the [official docs](https://go.dev/doc/security/fips140#fips-140-3-mode) correctly, having fips in the `on` mode, will just add some extra checks to the certain operations (tls package **ignores** certain protocols, rand is generated differently, etc). However, it will not error/panic when you do something "not complient". For that you have to put it into "fips only mode", for which you need a debug flag (`GODEBUG=fips140=only`).

> When GODEBUG=fips140=only is used, in addition to the above, cryptographic algorithms that are not FIPS 140-3 compliant will return an error or panic. Note that this mode is a best effort and can’t guarantee compliance with all FIPS 140-3 requirements.

Reply:
I don't know how necessary this `fips140=only` is, as it feels rather confusing to me that having the fips mode being `on` is not enough.

But still, we should at least run our e2e tests in this `only` mode.